custom/icds_reports/reports/poshan_progress_dashboard_data.py

```diff
@@ -18,6 +16,24 @@
 from custom.icds_reports.utils import apply_exclude, generate_quarter_months, calculate_percent, handle_average
 
 
+def fetch_month_data(value_fields, order_by, filters, include_test, domain):
+    queryset = PoshanProgressReportView.objects.filter(**filters).order_by(*order_by)
+    if not include_test:
+        queryset = apply_exclude(domain, queryset)
+    data = queryset.values(*value_fields)
+    return data
+
+
+def fetch_quarter_data(value_fields, order_by, filters, include_test, domain, months, data_period, unique_id):
```

Comment:
generic function to fetch the quarter data (reusing the month function), also prepare the quarter_dict(i.e. averaging values around the months). Doing it here as we need this in both comparative and aggregated structure.